sdv/sequential/par.py

```diff
@@ -151,6 +158,41 @@ def _validate_context_columns(self, data):
     def _validate(self, data):
         return self._validate_context_columns(data)
 
+    def _transform_sequence_index(self, data):
+        sequence_index = data[self._sequence_key + [self._sequence_index]]
+        sequence_index_context = sequence_index.groupby(self._sequence_key).agg('first')
+        sequence_index_context = sequence_index_context.rename(
+            columns={self._sequence_index: f'{self._sequence_index}.context'}
+        )
+        if all(sequence_index[self._sequence_key].nunique() == 1):
+            sequence_index_sequence = sequence_index[[self._sequence_index]].diff().bfill()
+        else:
+            sequence_index_sequence = sequence_index.groupby(self._sequence_key).apply(
+                lambda x: x[self._sequence_index].diff().bfill()
+            ).droplevel(1).reset_index()
+
+        if all(sequence_index_sequence[self._sequence_index].isna()):
+            fill_value = 0
+        else:
+            fill_value = min(sequence_index_sequence[self._sequence_index].dropna())
+        sequence_index_sequence = sequence_index_sequence.fillna(fill_value)
+
+        data[self._sequence_index] = sequence_index_sequence[self._sequence_index]
+        data = data.merge(
+            sequence_index_context,
+            left_on=self._sequence_key,
+            right_index=True)
```

Comment:
It's funny that we merge this back in but just separate it out again later. Not sure if there's a way to avoid that

Reply:
Maybe, but not easily right now. PAR also takes in the context columns when assembling the sequences and fitting the model so we'd need to do the join regardless. We could investigate if the context is really needed there though, and then skip the join here if so.